controllers/routemonitor/routemonitor_controller.go

```diff
@@ -25,55 +25,86 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client"
 
 	monitoringv1alpha1 "github.com/openshift/route-monitor-operator/api/v1alpha1"
+	utilreconcile "github.com/openshift/route-monitor-operator/pkg/util/reconcile"
 )
 
 // RouteMonitorReconciler reconciles a RouteMonitor object
+
 type RouteMonitorReconciler struct {
 	client.Client
 	Log    logr.Logger
 	Scheme *runtime.Scheme
+	// RouteMonitorInterface abstracts the Reconciler code from the controller resource
+	RouteMonitorInterface
```

Comment:
I think you don't need to add this here, as long as the struct implements all methods, it's compatible with the interface automatically. See https://gobyexample.com/interfaces

Reply:
will remove